src/patchelf.cc

```diff
@@ -1082,6 +1082,15 @@ void ElfFile<ElfFileParamNames>::modifySoname(sonameMode op, const std::string &
 
     debug("new SONAME is '%s'\n", newSoname.c_str());
 
+    /* If new SONAME fits in existing space, overwrite. This works around
+       issue #44. */
+    if (newSoname.size() <= sonameSize) {
+       debug("overwriting old SONAME with new...\n");
+       strcpy(soname, newSoname.c_str());
```

Comment:
Since we already know the size of `newSoname`, memcpy is the better choice:

```suggestion
       memcpy(soname, newSoname.c_str(), newSoname.size() + 1);
```